toqito/channel_props/is_extremal.py

```diff
@@ -0,0 +1,64 @@
+"""Determines whether a quantum channel is extremal."""
+
+import numpy as np
+from numpy.linalg import matrix_rank
+
+from toqito.channel_ops.choi_to_kraus import choi_to_kraus
+
+
+def is_extremal(channel):
```

Comment:
We should assume that `channel` is provided as either a list of Kraus operators or a matrix. We don't want to impose any additional structure on the input that has key values of strings. You can refer to the other functions in the `channel_props/` directory for reference.